### PR TITLE
Created documenation for Common PropTypes.

### PR DESCRIPTION
**Pull request by nickytonline**
## What type of PR is this? (check all applicable)

- [ ] Refactor
- [ ] Feature
- [ ] Bug Fix
- [ ] Optimization
- [x] Documentation Update

## Description

Adds documentation for PropTypes and our Common PropTypes.

## Related Tickets & Documents

#5882

## Added to documentation?

- [x] docs.dev.to
- [ ] readme
- [ ] no documentation needed

## [optional] What gif best describes this PR or how it makes you feel?

Sloth blinking

### CODE REVIEW

File: docs/frontend/preact.md [hunk 1]
```diff
@@ -12,3 +12,72 @@ Preact code is located in `app/javascript`.
 
 The components are mounted when needed, look for `javascript_pack_tag` in the
 view pages inside `app/views`.
+
+## PropTypes
+
+Preact supports
+[PropTypes](https://reactjs.org/docs/typechecking-with-proptypes.html). When
+creating Preact components, ensure to always have prop types defined.
```
Comment (vaidehijoshi) [inline docs/frontend/preact.md]:
```suggestion
creating Preact components, please ensure that you have defined your PropTypes.
```

**Comment by nickytonline** [comment]
> I think this adds some clarity, do we currently enforce proptypes with a linter? 

The linter does enforce it. For example, I've removed the PropTypes from the SearchForm.jsx component which results in a violation.

image

File: docs/frontend/preact.md [hunk 2]
```diff
@@ -12,3 +12,72 @@ Preact code is located in `app/javascript`.
...
+
+### Common PropTypes
+
+PropTypes can become repetitive. In simple cases where a PropType is a string or
+number, this duplication is expected. For commonly used entities in the DEV
+project, common PropTypes exist. All common PropTypes are located in the
+`/app/javascript/src/components/common-prop-types`. For example, the user entity
+that is referenced in many parts of the DEV codebase.
```
Comment (vaidehijoshi) [inline docs/frontend/preact.md]: (I can't make a GitHub suggestion, so I have to do this as a comment)

> PropTypes can become repetitive. In simple cases where a PropType is a string or
number, this duplication is expected. For commonly used entities in the DEV
project, common PropTypes exist. All common PropTypes are located in the
`/app/javascript/src/components/common-prop-types`. For example, the user entity
that is referenced in many parts of the DEV codebase.

Maybe this? _Using PropTypes can be repetitive. Some duplication is normal, like when a PropType is a string or a number. But for commonly-used PropTypes, like the `user` entity, you can use the provided common PropTypes, located in `/app/javascript/src/components/common-prop-types`, as shown below._

File: docs/frontend/preact.md [hunk 3]
```diff
@@ -12,3 +12,72 @@ Preact code is located in `app/javascript`.
...
+
+#### Using Common PropTypes
+
+Common PropTypes are imported just like any other
+[JavaScript Module](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Modules).
+For example, here are two scenarios where a component needs to use the
+`tagPropTypes`.
+
+Example 1: Using the `tagPropTypes` with `PropTypes.arrayOf` to construct the
+PropTypes for the `tags` prop.
```
Comment (vaidehijoshi) [inline docs/frontend/preact.md]: It might be nice to explain why someone might need to do this/when you'd want to do something like this.

Comment (nickytonline) [reply docs/frontend/preact.md]: I'll add a little blurb to explain. Thanks for all the great feedback! 🔥

**Pull request merged**